LoadingUISDK/LoadingUISDK/LoadingUIFramework/LoadingUIFramework/IndeterminateLoadingView.swift

```diff
@@ -0,0 +1,108 @@
+//
+//  IndeterminateLoadingView.swift
+//  LoadingUIFramework
+//
+//  Created by Matthew Martindale on 6/9/20.
+//  Copyright © 2020 Matthew Martindale. All rights reserved.
+//
+
+import UIKit
+
+class IndeterminateLoadingView: UIView, CAAnimationDelegate {
+
+    override init(frame: CGRect) {
+           super.init(frame: frame)
+           
+           setupShapeLayer()
+       }
+       
+       required init?(coder aDecoder: NSCoder) {
+           super.init(coder: aDecoder)
+           
+           setupShapeLayer()
+       }
+       
+       func startAnimating() {
+           guard !isAnimating else { return }
+           defer { isAnimating = true }
+       
+           startAnimation()
+       }
+       
+       func stopAnimating() {
+           guard isAnimating else { return }
+           
+           shouldStopAnimationOnNextCycle = true
+       }
+       
+       // MARK: - Private
+       
+       private func setupShapeLayer() {
+           let thickness: CGFloat = 10.0
+           
+           shapeLayer.frame = layer.bounds
+           shapeLayer.strokeColor = UIColor.black.cgColor
+           shapeLayer.fillColor = UIColor.clear.cgColor
+           shapeLayer.lineWidth = thickness
+           shapeLayer.strokeStart = 0.0
+           shapeLayer.strokeEnd = 0.0
+           layer.addSublayer(shapeLayer)
+           
+           let radius = min(bounds.width, bounds.height) / 2.0 - thickness/2.0
+           let rect = CGRect(x: bounds.midX - radius/2.0, y: bounds.midY - radius/2.0, width: radius, height: radius)
```

Comment:
No need to abbreviate `rect` here - be more clear as to what the rect refers to.